src/backup.py

```diff
@@ -3,18 +3,51 @@
 
 """Provides backup functionality for Synapse."""
 
+import datetime
 import logging
-from typing import Any, Optional
+import os
+import pathlib
+from typing import Any, Dict, Iterable, List, Optional
 
 import boto3
+import ops
 from botocore.config import Config
 from botocore.exceptions import BotoCoreError, ClientError
 from botocore.exceptions import ConnectionError as BotoConnectionError
+from ops.pebble import APIError, ExecError
 from pydantic import BaseModel, Field, validator
 
+import synapse
+
+AWS_COMMAND = "/aws/dist/aws"
+
+# The configuration files to back up consist in the signing keys
+# plus the sqlite db if it exists.
+BACKUP_FILE_PATTERNS = ["*.key", "homeserver.db*"]
+
+# For the data directory, inside the "media" directory, all directories starting
+# with local_ will be backed up. The directories starting with "remote_" are from
+# other server and is it not necessary to back them up.
+MEDIA_DIR = "media"
```

Comment:
This will be dynamic in a future PR and should be read from the homeserver.yaml file